app/assets/javascripts/govuk_publishing_components/lib/govspeak/youtube-link-enhancement.js

```diff
@@ -102,7 +104,8 @@
 
     // Only replace the <p> with a YT embed if the YT link is the only thing in the <p>.
     // This prevents other content in the <p> being lost.
-    if (parentPara.innerHTML !== $link.outerHTML) {
+    // However, if a <p> exists with only a YT link and punctuation, we do allow the punctuation characters to be lost to the YT embed.
+    if (parentPara.innerHTML.replaceAll(this.punctuationRegex, '') !== $link.outerHTML.replaceAll(this.punctuationRegex, '')) {
```

Comment:
If this line is duplicated maybe worth splitting off into a separate function?

Reply:
Good point, will move it to a function - thanks!